src/ray/raylet/node_manager.h

```diff
@@ -187,8 +187,20 @@ class NodeManager : public rpc::NodeManagerServiceHandler {
   ///
   /// \param task The task to fail.
   /// \param error_type The type of the error that caused this task to fail.
+  /// \param required_object_id The object id that we should write the error
+  ///        message into. This is used in case the task has no outputs (e.g.,
+  ///        it is the driver task).
   /// \return Void.
-  void TreatTaskAsFailed(const Task &task, const ErrorType &error_type);
+  void TreatTaskAsFailed(const Task &task, const ErrorType &error_type,
+                         const ObjectID *required_object_id = nullptr);
+  /// Mark the specified objects as failed with the given error type.
+  ///
+  /// \param error_type The type of the error that caused this task to fail.
+  /// \param object_ids The object ids to store error messages into.
+  /// \param job_id The optional job to push errors to if the writes fail.
+  void MarkObjectsAsFailed(const ErrorType &error_type,
+                           const std::vector<plasma::ObjectID> object_ids,
+                           const JobID *job_id = nullptr);
```

Comment:
Let's not use a default argument here since it's pretty scary if we don't have a `job_id`.